server/plugin.go

```diff
@@ -408,7 +408,7 @@ func (p *Plugin) OnActivate() error {
 		return err
 	}
 
-	recovery.Go("prefill_whitelist", p.API.LogError, func() {
+	recovery.Go("prefill_whitelist", p.API.LogError, p.GetMetrics(), func() {
```

Comment:
This is an example of why I don't understand the strategy here. Why is this store call special and needs a recovery, but the next p.start call does not? Why are they in goroutines at all? Can we really recover from whatever PrefillWhitelist is doing if it panics?

Reply:
This one is definitely up for debase as to whether it needed a goroutine in the first place. I think the only reason it's run asynchronously is because of the lock, but even then it's debatable. For now, I've left the underlying semantics of the code unchanged, but we're definitely reviewing all of this :)